# HG changeset patch
# User Punarbasu Purkayastha <ppurka@gmail.com>
# Date 1371232516 -28800
# Node ID c43c2cfae1e38a054d33ed98bc0046238d5ce258
# Parent  946962c0125fd6d493029ea58039e8e9d01b3508
fix parametric plots that fail due to invalid plot points

diff --git a/sage/plot/plot.py b/sage/plot/plot.py
--- a/sage/plot/plot.py
+++ b/sage/plot/plot.py
@@ -1237,6 +1237,11 @@
         1
         sage: p1.show(ymin=-10,ymax=10) # should be one legend
 
+    Parametric plots that get evaluated at invalid points should still
+    plot properly (:trac:`13246`)::
+
+        sage: parametric_plot((x, arcsec(x)), (x, -2, 2))
+
     """
 
     from sage.plot.misc import setup_for_eval_on_grid
@@ -1333,16 +1338,23 @@
         if abs(data[i+1][0] - data[i][0]) > 2*abs(xmax - xmin)/plot_points:
             exclude.append((data[i][0] + data[i+1][0])/2)
 
-    # We set exclude back to None if there are no points to be excluded
-    if exclude == []:
-        exclude = None
-    else:
-        exclude = sorted(exclude)
-
     if parametric:
         # We need the original x-values to be able to exclude points in parametric plots
         exclude_data = data
-        data = [(fdata, g(x)) for x, fdata in data]
+        newdata = []
+        for x,fdata in data:
+            try:
+                newdata.append((fdata, g(x)))
+            except ValueError:
+                newdata.append((fdata, 0)) # append a dummy value 0
+                exclude.append(x)
+        data = newdata
+
+    # We set exclude back to None if there are no points to be excluded
+    if not exclude:
+        exclude = None
+    else:
+        exclude.sort()
 
     G = Graphics()
 
